Sources/Controllers/2.0/Settings/NavigationSettings/OADefaultSpeedViewController.mm

```diff
@@ -137,58 +118,32 @@ - (void) viewDidLoad
 - (void) setupView
 {
     NSMutableArray *tableData = [NSMutableArray array];
-    NSMutableArray *minSpeedArr = [NSMutableArray array];
-    NSMutableArray *defaultSpeedArr = [NSMutableArray array];
-    NSMutableArray *maxSpeedArr = [NSMutableArray array];
-    [minSpeedArr addObject:@{
-        @"type" : kCellTypeSpeed,
-        @"title" : OALocalizedString(@"min_speed"),
-        @"value" : _minSpeedValue,
-    }];
-    [minSpeedArr addObject:@{
-        @"type" : kCellTypePicker,
-    }];
-    [defaultSpeedArr addObject:@{
+    if (_selectedValue == 0)
+        _selectedValue = _defaultValue;
+    
+    [tableData addObject:@{
         @"type" : kCellTypeSpeed,
         @"title" : OALocalizedString(@"default_speed"),
-        @"value" : _defaultSpeedValue,
-    }];
-    [defaultSpeedArr addObject:@{
-        @"type" : kCellTypePicker,
-    }];
-    [maxSpeedArr addObject:@{
-        @"type" : kCellTypeSpeed,
-        @"title" : OALocalizedString(@"max_speed"),
-        @"value" : _maxSpeedValue,
+        @"value" : [NSString stringWithFormat:@"%ld %@", _selectedValue, units],
     }];
-    [maxSpeedArr addObject:@{
-        @"type" : kCellTypePicker,
+    [tableData addObject:@{
+        @"type" : kCellTypeSlider,
+        @"minValue" : [NSString stringWithFormat:@"%ld %@", (long)_minValue, units],
+        @"maxValue" : [NSString stringWithFormat:@"%ld %@", (long)_maxValue, units],
     }];
-    if (![_speedParameters[@"defaultSpeedOnly"] boolValue])
-    {
-        [tableData addObject:minSpeedArr];
-        [tableData addObject:defaultSpeedArr];
-        [tableData addObject:maxSpeedArr];
-    }
-    else
-        [tableData addObject:defaultSpeedArr];
+    
     _data = [NSArray arrayWithArray:tableData];
 }
 
 - (IBAction)doneButtonPressed:(id)sender
 {
-    [_settings.defaultSpeed set:([_defaultSpeedValue floatValue] / _ratio) mode:_applicationMode];
-    if (![_speedParameters[@"defaultSpeedOnly"] boolValue])
-    {
-        [_settings.minSpeed set:([_minSpeedValue floatValue] / _ratio) mode:_applicationMode];
-        [_settings.maxSpeed set:([_maxSpeedValue floatValue] / _ratio) mode:_applicationMode];
-    }
+    [_settings.defaultSpeed set:(_selectedValue / _ratio) mode:self.appMode];
```

Comment:
You need to check if the mode is currently used by the RoutingHelper (has a calculated route) and if it is, call recalculateRouteDueToSettingsChange (see SettingsNavigationActivity.java line:787)